base_comment_template/tests/test_base_comment_template.py

```diff
@@ -1,37 +1,171 @@
+        )
+        self.assertEqual(templ, "")
+        templ = self.user.get_comment_template("after_lines")
+        self.assertEqual(templ, "Text after lines")
+
+    def test_partner_template_domain(self):
+        # Check getting the comment template if domain is set
+        self.before_template_id.partner_ids = self.partner2_id.ids
+        self.before_template_id.domain = "[('id', 'in', %s)]" % self.user.ids
+        templ = self.user.get_comment_template(
+            "before_lines", partner_id=self.partner2_id.id
+        )
+        self.assertEqual(templ, "Text before lines")
+        templ = self.user2.get_comment_template(
+            "before_lines", partner_id=self.partner_id.id
+        )
+        self.assertEqual(templ, "")
+
+    def test_company_partner_template_domain(self):
+        # Check getting the comment template with company and if domain is set
+        self.before_template_id.company_id = self.company
+        templ = self.user.get_comment_template("before_lines")
+        self.assertEqual(templ, "")
+        templ = self.user.get_comment_template(
+            "before_lines", company_id=self.company.id
+        )
+        self.assertEqual(templ, "Text before lines")
+        self.before_template_id.partner_ids = self.partner2_id.ids
+        self.before_template_id.domain = "[('id', 'in', %s)]" % self.user.ids
+        templ = self.user.get_comment_template(
+            "before_lines", partner_id=self.partner2_id.id
+        )
+        self.assertEqual(templ, "")
+        self.before_template_id.company_id = self.env.user.company_id
+        templ = self.user.get_comment_template(
+            "before_lines", partner_id=self.partner2_id.id
+        )
+        self.assertEqual(templ, "Text before lines")
+        templ = self.user2.get_comment_template(
+            "before_lines", partner_id=self.partner2_id.id
+        )
+        self.assertEqual(templ, "")
 
-    def test_commercial_partner_fields(self):
-        # Azure Interior
-        partner_id = self.env.ref("base.res_partner_12")
-        partner_id.property_comment_template_id = self.template_id.id
-        # Test childs propagation of commercial partner field
-        for child_id in partner_id.child_ids:
-            self.assertEqual(child_id.property_comment_template_id, self.template_id)
+    def test_priority(self):
+        # Check setting default template will change previous record default
+        new_template = self.env["base.comment.template"].create(
+            {
+                "name": "before_lines",
+                "text": "Text before lines 1",
+                "model_ids": [(6, 0, self.user_obj.ids)],
+                "priority": 2,
+            }
+        )
 
-    def test_get_value_without_partner(self):
-        self.assertEqual(self.template_id.get_value(), "<p>Text before lines</p>")
+        self.assertEqual(new_template.text, "Text before lines 1")
 
-    def test_get_value_with_partner(self):
```

Comment:
While migration I've had that test to improve the coverage, here you get a nice coverage regarding that code but could be nice to ensure lang param is properly forward by using a different language and asserting expected result, I haven't seen such test case kept.

Reply:
Now, it's not necessary.